src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -294,11 +294,10 @@ public TagHelperContent EndTagHelperWritingScope()
 
                 foreach (var attribute in tagHelperOutput.Attributes)
                 {
-                    var value = HtmlEncoder.HtmlEncode(attribute.Value);
                     writer.Write(' ');
                     writer.Write(attribute.Key);
                     writer.Write("=\"");
-                    writer.Write(value);
+                    WriteTo(writer, HtmlEncoder, attribute.Value, escapeQuotes: true);
```

Comment:
no. `WriteTo` is the helper.  otherwise end up special-casing `HtmlString` in two layers.

the real fix is to know how attribute values were quoted in the original Razor source.  until we have that, this workaround is sufficient.